### PR TITLE
chore(#8241): migrate africas-talking e2e test

### PR DESCRIPTION
**Pull request by tatilepizs**
# Description

Migrate the africas-talking e2e test from protractor to wdio.

This test covers the following scenarios:

- africas talking api  - gateway submits new WT sms messages
- africas talking api  - gateway submits WT sms status updates

medic/cht-core#8241

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:migrate-africas-talkking/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:migrate-africas-talkking/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:migrate-africas-talkking/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/default/sms/africas-talking.wdio-spec.js [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+    const messageTo1 = '+64275555556';
+    const messageTo2 = '+64275555557';
+    const messageTo3 = '+64275555558';
+    const messageContent1 = 'Thank you for registering Shannon. Their pregnancy ID is 28551, and EDD is ' +
+      'Sun, Dec 18th, 2016';
+    const messageContent2 = 'Please remind Shannon (28551) to visit the health facility for ANC visit this week. ' +
+      'When she does let us know with "V 28551". Thanks!';
+    const messageContent3 = 'Did Shannon attend her ANC visit? When she does, respond with "V 28551". Thank you!';
+
+    const report = {
```
Comment (lorerod) [inline tests/e2e/default/sms/africas-talking.wdio-spec.js:72]: What do you think about using [sms-pregnancy](https://github.com/medic/cht-core/blob/master/tests/factories/cht/reports/sms-pregnancy.js) factory? You may need to change the original factory and specify [options](https://github.com/rosiejs/rosie#programmatic-generation-of-attributes) to generate the attributes.
As an example, I use it in a couple of factories [here](https://github.com/medic/cht-core/tree/master/tests/factories/brac).

**Review by lorerod — CHANGES_REQUESTED** [review]
Thanks, @tatilepizs, for this. We are one step closer to finishing the protractor migration!
I left you a suggestion. Let me know what you think.

File: tests/page-objects/default/sms/messages.wdio.page.js [hunk 2]
```diff
@@ -50,6 +50,7 @@ const getMessageContent = async  (index = 1) => {
   return {
     content: await sms.$('p[test-id="sms-content"]').getText(),
     state: await sms.$('.state').getText(),
+    dataID: await sms.getAttribute('data-id'),
```
Comment (jkuester) [inline tests/page-objects/default/sms/messages.wdio.page.js:53]:
```suggestion
    dataId: await sms.getAttribute('data-id'),
```

Super minor, but usually I see it like this when camel-casing "ID".

File: tests/e2e/default/sms/africas-talking.wdio-spec.js [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+      const { name } = await messagesPage.getMessageHeader();
+      expect(name).to.equal(rawNumber);
+
+      const { content, state, dataID } = await messagesPage.getMessageContent(1);
+      expect(content).to.equal(message);
+      expect(state).to.equal('received');
+
+      const doc = await utils.getDoc(dataID);
+      expect(doc.sms_message && doc.sms_message.gateway_ref).to.equal('messageID');
```
Comment (jkuester) [inline tests/e2e/default/sms/africas-talking.wdio-spec.js:53]:
```suggestion
      expect(doc.sms_message.gateway_ref).to.equal('messageID');
```

IDK if I am missing something here perhaps, but I guess this assertion will fail if `doc.sms_message` exists or not.  I think it is a lot easier to read if you do not bother with the extra code here....

**Review by jkuester — APPROVED** [review]
Tres bien!  LGTM!

File: tests/e2e/default/sms/africas-talking.wdio-spec.js (hunk 1 above)
Comment (tatilepizs) [reply tests/e2e/default/sms/africas-talking.wdio-spec.js]: That was a great idea Lore, thank you for helping in the process! 🤍

**Review by lorerod — APPROVED** [review]
Thanks for implementing my suggestion.

**Pull request merged**